lib/source/pl/lib/std/pragmas.cpp

```diff
@@ -21,9 +37,10 @@ namespace pl::lib::libstd {
         });
 
         runtime.addPragma("eval_depth", [](pl::PatternLanguage &runtime, const std::string &value) {
-            auto limit = strtol(value.c_str(), nullptr, 0);
+            long unsigned int limit = 0;
+            bool parsed = parse_limit(value, &limit);
 
-            if (limit <= 0)
+            if (!parsed || limit == LIMITMAX)
```

Comment:
When you use an optional for the function above, it might look something like 

```cpp
if (auto limit = parseLimit(value); !limit.has_value())
    return false;
else
    runtime.getInternals().evaluator->setEvaluationDepth(limit.value());
```